React/Views/RCTView.m

```diff
@@ -1566,25 +1566,30 @@ - (BOOL)performDragOperation:(id <NSDraggingInfo>)sender
 
 #pragma mark - Keyboard Events
 
-#if TARGET_OS_OSX // TODO: add iOS keyboard event handling
+#if TARGET_OS_OSX
 - (void)keyDown:(NSEvent *)event {
-  if (!self.onKeyDown) {
+  if (self.onKeyDown != nil) {
     [super keyDown:event];
     return;
   }
 
-  RCTViewKeyboardEvent *keyboardEvent = [RCTViewKeyboardEvent keyDownEventWithReactTag:self.reactTag characters:event.characters modifier:event.modifierFlags];
-  [_eventDispatcher sendEvent:keyboardEvent];
+  // only post events for keys we care about
+  if ([[self validKeysDown] containsObject:event.characters]) {
```

Comment:
if this is not contained then we don't call super and nobody gets the event?

Reply:
i.e. back to the original comment - the presence of an onKeyDown will mean this view eats all key events and doesn't let any of them pass up the responder chain?